prisma-filter/src/filter.parser.ts

```diff
@@ -120,6 +132,10 @@ export class FilterParser<TDto, TWhereInput> {
       throw new Error(`Filter value must be a string, a number or a boolean`);
     }
 
+    if (type === FilterOperationType.ArrayContains) {
+      return [rawValue] as string[] | number[] | boolean[];
```

Comment:
Here, you also have to differentiate between `ArrayContains` and `ArrayContainsStrings`. `ArrayContains` has to potentially cast the value to a boolean or number (like the code below does), while `ArrayContainsStrings` can directly return the `rawValue`.